src/tainting/Dataflow_tainting.ml

```diff
@@ -637,21 +638,41 @@ let find_pos_in_actual_args args_taints fparams =
     taint_opt
 
 let fix_poly_taint_with_field env lval st =
-  if env.lang =*= Lang.Java || Lang.is_js env.lang then
-    match lval.rev_offset with
-    | { o = Dot n; _ } :: _ -> (
-        match st with
-        | `Sanitized
-        | `Clean
-        | `None ->
-            st
-        | `Tainted taints -> (
-            match !(n.id_info.id_type) with
-            | Some { t = TyFun _; _ } ->
+  let type_of_il_offset il_offset =
+    match il_offset.IL.o with
+    | Dot n -> !(n.id_info.id_type)
+    | Index _ -> None
+  in
+  let arg_offset_of_il_offset il_offset =
+    match !hook_arg_offset_of_il_offset with
+    | None -> (
+        match il_offset.IL.o with
+        | Dot n -> Some (T.Ofld n)
+        | Index _ ->
+            (* no index-sensitivity in OSS *)
+            None)
+    | Some arg_offset_of_il_offset -> arg_offset_of_il_offset il_offset
+  in
+  (* TODO: Aren't we missing here C# and Go ? *)
+  if env.lang =*= Lang.Java || Lang.is_js env.lang || env.lang =*= Lang.Python
```

Comment:
Would it make sense to check the list of pro-languages to keep this in sync as we add new languages to pro?

Reply:
Nice idea, in principle we could even just do it for every language, but need to check if anything breaks.

Reply:
I know you mean Pro-supported languages, because by "Pro language" we typically mean languages _only_ available in Pro like Apex and Elixir.

Reply:
Let me put this in a separate PR just in case...